internal/pkg/service/stream/storage/level/local/diskwriter/router/connection/connection.go

```diff
@@ -114,8 +113,8 @@ func NewManager(ctx context.Context, d dependencies, cfg network.Config, nodeID
 			WithOnUpdate(func(_ etcdop.MirrorUpdatedKeys[*volumeData]) {
 				m.updateConnections(ctx)
 			}).
-			StartMirroring(ctx, &m.wg)
-		if err := <-errCh; err != nil {
+			Build()
```

Comment:
Here is the issue:
- `WithOnUpdate` - `m.updateConnections` is using `m.volumes`.
- The on update callback is triggered by a background etcd watcher
- So Go detects a dara race here.

![image](https://github.com/user-attachments/assets/2dcb2be0-ec32-4211-8f8f-8b5390018d2f)